File: examples/modulith-clinical-tool-jpa/src/main/java/io/zenwave360/example/clinicaltool/modules/surveys/service/SurveysService.java
package io.zenwave360.example.clinicaltool.modules.surveys.service;

import io.zenwave360.example.clinicaltool.modules.surveys.domain.*;
import io.zenwave360.example.clinicaltool.modules.surveys.service.dtos.*;
import java.math.*;
import java.time.*;
import java.util.*;

/**
 * Inbound Service Port for managing [SurveyAnswers].
 */
@org.springframework.modulith.NamedInterface("SurveysService")
public interface SurveysService {

    /**
     *
     *
     */
    public SurveyAndQuestions getSurveyAndQuestionsForPatient(String name, Long patientId, String lang);
    /**
     *
     *
     */
    public SurveyAnswers answerSurvey(Long surveyId, Long patientId, LocalDate date, SurveyAnswers input);
    /**
     *
     *
     */
    public Optional<SurveyAnswers> updateSurveyAnswers(
            Long surveyId, Long patientId, LocalDate date, java.util.Map input);
    /**
     *
     *
     */
    public Optional<SurveyAnswers> getSurveyAnswers(Long surveyId, Long patientId, LocalDate date);
}

File: examples/modulith-klinical-tool-jpa/src/main/kotlin/io/zenwave360/example/clinicaltool/modules/surveys/service/SurveysService.kt
package io.zenwave360.example.clinicaltool.modules.surveys.service

import io.zenwave360.example.clinicaltool.modules.surveys.domain.*
import io.zenwave360.example.clinicaltool.modules.surveys.service.dtos.*
import java.math.*
import java.time.*

/** Inbound Service Port for managing [SurveyAnswers]. */
@org.springframework.modulith.NamedInterface("SurveysService")
interface SurveysService {

    /**  */
    fun getSurveyAndQuestionsForPatient(name: String, patientId: Long, lang: String?): SurveyAndQuestions

    /**  */
    fun answerSurvey(surveyId: Long, patientId: Long, date: LocalDate, input: SurveyAnswers): SurveyAnswers

    /**  */
    fun updateSurveyAnswers(surveyId: Long, patientId: Long, date: LocalDate, input: Map<String, Any?>): SurveyAnswers?

    /**  */
    fun getSurveyAnswers(surveyId: Long, patientId: Long, date: LocalDate): SurveyAnswers?
}
